src/localplayer.cpp

```diff
@@ -726,354 +72,52 @@ v3f LocalPlayer::getEyeOffset() const
-			f32 max_axis_distance_f = MYMAX(
-					std::fabs(player_p2df.X - node_p2df.X),
-					std::fabs(player_p2df.Y - node_p2df.Y));
-
-			if (distance_f > min_distance_f ||
-					max_axis_distance_f > 0.5 * BS + sneak_max + 0.1 * BS)
-				continue;
-
-			// The node to be sneaked on has to be walkable
-			node = map->getNodeNoEx(p, &is_valid_position);
-			if (!is_valid_position || !nodemgr->get(node).walkable)
-				continue;
-			// And the node above it has to be nonwalkable
-			node = map->getNodeNoEx(p + v3s16(0, 1, 0), &is_valid_position);
-			if (!is_valid_position || nodemgr->get(node).walkable)
-				continue;
-			// If not 'sneak_glitch' the node 2 nodes above it has to be nonwalkable
-			if (!physics_override_sneak_glitch) {
-				node =map->getNodeNoEx(p + v3s16(0, 2, 0), &is_valid_position);
-				if (!is_valid_position || nodemgr->get(node).walkable)
-					continue;
-			}
-
-			min_distance_f = distance_f;
-			new_sneak_node = p;
-		}
-
-		bool sneak_node_found = (min_distance_f < 100000.0 * BS * 0.9);
-
-		m_sneak_node = new_sneak_node;
-		m_sneak_node_exists = sneak_node_found;
-
-		if (sneak_node_found) {
-			f32 cb_max = 0;
-			MapNode n = map->getNodeNoEx(m_sneak_node);
-			std::vector<aabb3f> nodeboxes;
-			n.getCollisionBoxes(nodemgr, &nodeboxes);
-			for (const auto &box : nodeboxes) {
-				if (box.MaxEdge.Y > cb_max)
-					cb_max = box.MaxEdge.Y;
-			}
-			m_sneak_node_bb_ymax = cb_max;
-		}
-
-		/*
-			If sneaking, the player's collision box can be in air, so
-			this has to be set explicitly
-		*/
-		if (sneak_node_found && control.sneak)
-			touching_ground = true;
-	}
-
-	/*
-		Set new position but keep sneak node set
-	*/
-	bool sneak_node_exists = m_sneak_node_exists;
-	setPosition(position);
-	m_sneak_node_exists = sneak_node_exists;
-
-	/*
-		Report collisions
-	*/
-	// Dont report if flying
-	if (collision_info && !(player_settings.free_move && fly_allowed)) {
-		for (const auto &info : result.collisions) {
-			collision_info->push_back(info);
-		}
-	}
+	return m_client ? m_client->ndef() : NULL;
+}
 
-	if (!result.standing_on_object && !touching_ground_was && touching_ground) {
-		m_client->getEventManager()->put(new SimpleTriggerEvent(MtEvent::PLAYER_REGAIN_GROUND));
-		// Set camera impact value to be used for view bobbing
-		camera_impact = getSpeed().Y * -1;
-	}
+void LocalPlayer::handleAttachedMove()
+{
+	setPosition(overridePosition);
+}
 
-	{
-		camera_barely_in_ceiling = false;
-		v3s16 camera_np = floatToInt(getEyePosition(), BS);
-		MapNode n = map->getNodeNoEx(camera_np);
-		if (n.getContent() != CONTENT_IGNORE) {
-			if (nodemgr->get(n).walkable && nodemgr->get(n).solidness == 2)
-				camera_barely_in_ceiling = true;
-		}
-	}
+float LocalPlayer::getStepHeight() const
+{
+	// Player object property step height is multiplied by BS in
+	// /src/script/common/c_content.cpp and /src/content_sao.cpp
+	return (m_cao == nullptr)
+			       ? 0.0f
+			       : (touching_ground ? m_cao->getStepHeight() : (0.2f * BS));
```

Comment:
wrong indentation
also this would look much better as an `if`

Reply:
Fixed.